src/main/java/seedu/address/ui/EventCard.java

```diff
@@ -38,18 +43,59 @@ public class EventCard extends UiPart<Region> {
     @FXML
     private FlowPane attendees;
 
+    @FXML
+    private ImageView calendarIcon;
+
+    @FXML
+    private ImageView personIcon;
+
     /**
      * Creates a {@code EventCard} with the given {@code Event} and index to display.
      */
     public EventCard(Event event, int displayedIndex) {
         super(FXML);
         this.event = event;
         id.setText(displayedIndex + ". ");
-        name.setText(event.getEventName());
-        date.setText(event.getDate().toString());
-        event.getAttendees().stream()
+        name.setText("  " + event.getEventName());
+        date.setText("  " + event.getDate().format(DateTimeFormatter.ofPattern(" MMM dd, yyyy")));
+        List<Person> sortedAttendees = event.getAttendees().stream()
                 .sorted(Comparator.comparing(person -> person.getName().toString()))
-                .forEach(person ->
-                        attendees.getChildren().add(new Label(person.getName().toString())));
+                .toList();
+
+        int maxDisplay = 2;
+        int numOfAttendees = sortedAttendees.size();
+        attendees.getChildren().add(new Label("  "));
+        for (int i = 0; i < maxDisplay; i++) {
+            String name = sortedAttendees.get(i).getName().toString();
+            String formattedName = (i == maxDisplay - 1)
+                                   ? name
+                                   : name + ", ";
+            attendees.getChildren().add(new Label(formattedName));
+        }
+
+        if (numOfAttendees > maxDisplay) {
+            int remainingPeople = numOfAttendees - maxDisplay;
+            attendees.getChildren().add(new Label(", and " + remainingPeople + " more"));
+        }
+
+    }
+
+    /**
+     * Toggles the icons for the {@code calendarIcon} and {@code personIcon} based on selection state of event card.
+     *
+     * If the event card is selected, then the dark-themed versions of the icons are displayed.
+     * Else, the light versions of the icons are displayed.
+     *
+     * @param selected {@code true} if the card is selected.
+     *                 {@code false} if the card is not selected.
+     */
+    public void toggleIcons(boolean selected) {
```

Comment:
This piece of code is creating new images on every toggle, might be a bit inefficient. Maybe cache the images and reuse the same object instead of creating new ones during every single toggle.

Reply:
Agreed, I'll make the changes and update the pr.